src/common/commonutils/PassUtils.c

```diff
@@ -145,25 +280,103 @@ int SetLockoutForFailedPasswordAttempts(void* log)
     const char* pamTally2Line = "auth required pam_tally2.so file=/var/log/tallylog onerr=fail audit silent deny=5 unlock_time=900 even_deny_root\n";
     const char* pamFailLockLine = "auth required [default=die] pam_faillock.so preauth silent audit deny=3 unlock_time=900 even_deny_roo\nt";
     const char* etcPamdLogin = "/etc/pam.d/login";
-    const char* etcPamdSystemAuth = "/etc/pam.d/system-auth"; 
     const char* etcPamdPasswordAuth = "/etc/pam.d/password-auth";
+    const char* etcPamdLoginCopy = "/etc/pam.d/~login.copy";
+    const char* g_etcPamdSystemAuthCopy = "/etc/pam.d/~system-auth.copy";
+    const char* etcPamdPasswordAuthCopy = "/etc/pam.d/~password-auth.copy";
     const char* marker = "auth";
+    char* original = NULL;
+    int status = ENOENT, _status = ENOENT;
 
-    int status = ENOENT;
-
-    if (0 == CheckFileExists(etcPamdSystemAuth, NULL, log))
+    if (0 == CheckFileExists(g_etcPamdSystemAuth, NULL, log))
     {
-        status = ReplaceMarkedLinesInFile(etcPamdSystemAuth, marker, pamFailLockLine, '#', log);
```

Comment:
why are we expanding this function? Were there issues?

Reply:
As the title and PR descriptions say.... Should I quote here? Do you understand the difference between what was before one one line and what it is now? Rather than replacing/adding lines in place to file, make a temporary copy, change that one and when done atomically move that temp file to the target. Simple as that.